compiler-core/src/language_server/engine.rs

```diff
@@ -360,7 +359,12 @@ where
                 if !type_.public {
                     continue;
                 }
-                completions.push(type_completion(Some(&alias), name, type_));
+                let module = match import.clone().as_name {
+                    Some((AssignName::Variable(name), _)) => Some(name),
+                    Some((AssignName::Discard(_), _)) => None,
```

Comment:
This is registering types as being in scope unqualified when they are not actually in scope. Unqualified types are handled in the block below.